demo/src/main/java/com/google/maps/android/utils/demo/ClusteringDemoActivity.java

```diff
@@ -43,6 +48,32 @@ protected void startDemo(boolean isRestore) {
         mClusterManager = new ClusterManager<>(this, getMap());
         getMap().setOnCameraIdleListener(mClusterManager);
 
+        // Add a custom InfoWindowAdapter by setting it to the MarkerManager.Collection object from
+        // ClusterManager rather than from GoogleMap.setInfoWindowAdapter
+        mClusterManager.getMarkerCollection().setInfoWindowAdapter(new GoogleMap.InfoWindowAdapter() {
+            @Override
+            public View getInfoWindow(Marker marker) {
+                final LayoutInflater inflater = LayoutInflater.from(ClusteringDemoActivity.this);
+                final View view = inflater.inflate(R.layout.custom_info_window, null);
+                final TextView textView = view.findViewById(R.id.textViewTitle);
+                String text = (marker.getTitle() != null) ? marker.getTitle() : "Cluster Item";
+                textView.setText(text);
+                return view;
+            }
+
+            @Override
+            public View getInfoContents(Marker marker) {
+                return null;
+            }
+        });
+        mClusterManager.getMarkerCollection().setOnInfoWindowClickListener(new GoogleMap.OnInfoWindowClickListener() {
```

Comment:
Nit (kind of) - we could add Java 8 support to the demo app and use lambdas.

Generally, though, it may be a good idea to add Java 8 support to the app anyway to help developers avoid this issue - https://github.com/googlemaps/android-maps-utils/issues/741#issuecomment-638469115 - if they are copying some of the project properties, or just looking at the `build.gradle`.

Reply:
Good idea! Let me do that in a separate PR.